#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_goffice_0-8_support.dpatch by Jean-Louis Dupond <info@dupondje.be>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Allowing gnucash to compile with goffice 0-8 (taken from diff 18311 in gnucash).

@DPATCH@
diff -urNad gnucash-2.2.9~/src/gnome-utils/gnc-html-graph-gog.c gnucash-2.2.9/src/gnome-utils/gnc-html-graph-gog.c
--- gnucash-2.2.9~/src/gnome-utils/gnc-html-graph-gog.c	2008-01-08 02:06:26.000000000 +0100
+++ gnucash-2.2.9/src/gnome-utils/gnc-html-graph-gog.c	2010-02-13 00:25:40.973428634 +0100
@@ -48,6 +48,9 @@
 #endif
 #include <goffice/graph/gog-style.h>
 #include <goffice/graph/gog-styled-object.h>
+#ifndef GO_COLOR_FROM_GDK 
+#       define GO_COLOR_FROM_GDK GDK_TO_UINT 
+#endif
 #include <goffice/graph/gog-plot.h>
 #include <goffice/graph/gog-series.h>
 #include <goffice/utils/go-color.h>
@@ -363,8 +366,13 @@
   }
   gog_object_add_by_name(chart, "Legend", NULL);
 
+#ifdef GO_COLOR_BLACK
+  GOG_STYLED_OBJECT(graph)->style->line.width = 5;
+  GOG_STYLED_OBJECT(graph)->style->line.color = GO_COLOR_BLACK;
+#else
   GOG_STYLED_OBJECT(graph)->style->outline.width = 5;
   GOG_STYLED_OBJECT(graph)->style->outline.color = RGBA_BLACK;
+#endif
 
   series = gog_plot_new_series(plot);
   labelData = go_data_vector_str_new((char const * const *)labels, datasize, NULL);
@@ -493,7 +501,7 @@
       style->fill.type = GOG_FILL_STYLE_PATTERN;
       if (gdk_color_parse (col_colors[i], &color)) {
            style->fill.auto_back = FALSE;
-           go_pattern_set_solid (&style->fill.pattern, GDK_TO_UINT (color));
+           go_pattern_set_solid (&style->fill.pattern, GO_COLOR_FROM_GDK(color));
       } else {
            g_warning("cannot parse color [%s]", col_colors[i]);
       }
@@ -588,9 +596,9 @@
     GdkColor color;
     if (gdk_color_parse(color_str, &color)) {
       style->marker.auto_outline_color = FALSE;
-      go_marker_set_outline_color(style->marker.mark, GDK_TO_UINT(color));
+      go_marker_set_outline_color(style->marker.mark, GO_COLOR_FROM_GDK(color));
       style->line.auto_color = FALSE;
-      style->line.color = GDK_TO_UINT(color);
+      style->line.color = GO_COLOR_FROM_GDK(color);
     } else {
       g_warning("cannot parse color [%s]", color_str);
     }
